libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -240,6 +243,8 @@ public void run() {
             }, t.getDuration());
             t.show();
         }
+        final Intent intent = new Intent(AUTHENTICATION_FAILED_INTENT);
```

Comment:
@csoon03 I can pass in `error` and `errorDesc` if you like. `error` is a `String` that represents the error like `access_denied`, for instance, and `errorDesc` is a human readable description of that. Typically, these fields are set for errors when the request fails that are returned from the server. If nothing is returned from the server, these fields might be empty. Does that work for you?